### PR TITLE
Fixed #32357 -- Dropped support for PostgreSQL 9.6 and PostGIS 2.3.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-32357

### CODE REVIEW

**Review by claudep — APPROVED** [review]
I'm fine with those removals. Thanks!

File: docs/ref/contrib/gis/install/index.txt [hunk 1]
```diff
@@ -347,7 +347,7 @@ running macOS.  Because MacPorts still builds the software from source,
 
 Summary::
 
-    $ sudo port install postgresql93-server
+    $ sudo port install postgresql10-server
```
Comment (pope1ni) [inline docs/ref/contrib/gis/install/index.txt:350]: Would it make sense to make this the latest available release, i.e. 13? If users are starting out it feels like they should be pointed to the latest stable release where possible.

File: docs/ref/databases.txt [hunk 2]
```diff
@@ -103,7 +103,7 @@ below for information on how to set up your database correctly.
 PostgreSQL notes
 ================
 
-Django supports PostgreSQL 9.6 and higher. `psycopg2`_ 2.5.4 or higher is
+Django supports PostgreSQL 10 and higher. `psycopg2`_ 2.5.4 or higher is
```
Comment (pope1ni) [inline docs/ref/databases.txt:106]: We probably ought to check this version of `psycopg2` as it is very old. With supporting Python 3.8 as a minimum in 4.0 this is probably not right.

Obviously separate PR - just putting it on the radar. I was giving some thought to minimum versions in ticket-28905, #10663 - probably something to pick up there.

File: docs/ref/unicode.txt [hunk 3]
```diff
@@ -18,8 +18,8 @@ able to store certain characters in the database, and information will be lost.
 * MySQL users, refer to the `MySQL manual`_ for details on how to set or alter
   the database character set encoding.
 
-* PostgreSQL users, refer to the `PostgreSQL manual`_ (section 22.3.2 in
-  PostgreSQL 9) for details on creating databases with the correct encoding.
+* PostgreSQL users, refer to the `PostgreSQL manual`_ (section 23.3.2 in
+  PostgreSQL 10) for details on creating databases with the correct encoding.
```
Comment (pope1ni) [inline docs/ref/unicode.txt:22]: Ditto for 10 → 13?

**Review by pope1ni — APPROVED** [review]
Looks good.

Just had a couple of thoughts on the documentation.

File: docs/ref/databases.txt (hunk 2 above)
Comment (felixxm) [reply docs/ref/databases.txt:106]: Normally we don't bump drivers versions without a specific reason, e.g. `2.5.4` added support for `JSONB`. If you think it's worth bumping or support is already broken, please create a ticket with rationale.

File: docs/ref/contrib/gis/install/index.txt (hunk 1 above)
Comment (felixxm) [reply docs/ref/contrib/gis/install/index.txt:350]: As far as I'm aware `13` is not available on MacPorts.

File: docs/ref/unicode.txt (hunk 3 above)
Comment (felixxm) [reply docs/ref/unicode.txt]: I've changed the link.

File: docs/ref/contrib/gis/install/index.txt (hunk 1 above)
Comment (pope1ni) [reply docs/ref/contrib/gis/install/index.txt:350]: Yeah. Looks like 12 is the latest on MacPorts, so we should probably go with that. Homebrew has 13. But IANAMU* so :man_shrugging: .

\* = I am not a Mac user. :joy:

Comment (felixxm) [reply docs/ref/contrib/gis/install/index.txt:350]: me neither :shrug:

Comment (pauloxnet) [reply docs/ref/contrib/gis/install/index.txt]: I'm not a Mac User but I can see PostgreSQL 13 on MacPorts
https://ports.macports.org/port/postgresql13/

Comment (felixxm) [reply docs/ref/contrib/gis/install/index.txt]: Let's leave it. I will confirm with Carlton in the next week.

File: docs/ref/contrib/gis/install/geolibs.txt [hunk 4]
```diff
@@ -12,7 +12,7 @@ Program                   Description                           Required
 `PROJ`_                   Cartographic Projections library      Yes (PostgreSQL and SQLite only)  6.x, 5.x, 4.x
 :doc:`GDAL <../gdal>`     Geospatial Data Abstraction Library   Yes                               3.1, 3.0, 2.4, 2.3, 2.2, 2.1, 2.0
 :doc:`GeoIP <../geoip2>`  IP-based geolocation library          No                                2
-`PostGIS`__               Spatial extensions for PostgreSQL     Yes (PostgreSQL only)             3.0, 2.5, 2.4, 2.3
+`PostGIS`__               Spatial extensions for PostgreSQL     Yes (PostgreSQL only)             3.0, 2.5, 2.4
```
Comment (pauloxnet) [inline docs/ref/contrib/gis/install/geolibs.txt:15]: Can we add PostGIS 3.1 ?

Comment (pope1ni) [reply docs/ref/contrib/gis/install/geolibs.txt:15]: That'll be handled in a separate PR.

Comment (felixxm) [reply docs/ref/contrib/gis/install/geolibs.txt:15]: No in this PR, that is a separate issue not related with dropping support for PostGIS 2.3. Feel-free to create a ticket.

Comment (pauloxnet) [reply docs/ref/contrib/gis/install/geolibs.txt:15]: ok thanks

File: docs/ref/contrib/gis/install/index.txt (hunk 1 above)
Comment (carltongibson) [reply docs/ref/contrib/gis/install/index.txt]: IANAM**P**U 😄 

As per @paulxnet's https://github.com/django/django/pull/13894#discussion_r558183709 looks like it's there or will be. 
(I'd imagine folks can work it out…) 

👍 from me to the general idea of the pointing to latest version. (No doubt it'll be another few years before anyone updates this again so...)

**Pull request merged**